modulemd/v2/modulemd-defaults-v1.c

```diff
@@ -46,6 +47,105 @@ G_DEFINE_TYPE (ModulemdDefaultsV1,
                MODULEMD_TYPE_DEFAULTS)
 
 
+static gboolean
+modulemd_defaults_v1_equals (ModulemdDefaults *self_1, ModulemdDefaults *self_2, GError **error)
+{
+  ModulemdDefaultsV1 *v1_self_1 = NULL;
+  ModulemdDefaultsV1 *v1_self_2 = NULL;
+  g_autoptr (GError) nested_error = FALSE;
+  g_autoptr (GPtrArray) profile_defaults_1 = NULL;
+  g_autoptr (GPtrArray) profile_defaults_2 = NULL;
+  g_autoptr (GPtrArray) intent_default_profiles_1 = NULL;
+  g_autoptr (GPtrArray) intent_default_profiles_2 = NULL;
+  gchar *key_1 = NULL;
+  gchar *key_2 = NULL;
+  GHashTable * table_1 = NULL;
+  GHashTable * table_2 = NULL;
+  int i = 0;
+
+  g_return_val_if_fail (MODULEMD_IS_DEFAULTS_V1 (self_1), FALSE);
+  v1_self_1 = MODULEMD_DEFAULTS_V1 (self_1);  
+  g_return_val_if_fail (MODULEMD_IS_DEFAULTS_V1 (self_2), FALSE);
+  v1_self_2 = MODULEMD_DEFAULTS_V1 (self_2);  
+
+  if (!MODULEMD_DEFAULTS_CLASS (modulemd_defaults_v1_parent_class)
+             ->equals (self_1, self_2, error))
+    {
+      g_propagate_error (error, g_steal_pointer (&nested_error));
+      return FALSE;
+    }
+
+  if(g_strcmp0 (v1_self_1->default_stream, v1_self_2->default_stream) != 0)
+    {
+      return FALSE;
+    } 
+
+  //Check profile_defaults: size, keys, values
+  if (g_hash_table_size (v1_self_1->profile_defaults) != g_hash_table_size(v1_self_2->profile_defaults))
+    {
+      return FALSE;
+    }
+  profile_defaults_1 = modulemd_ordered_str_keys (v1_self_1->profile_defaults, modulemd_strcmp_sort);
+  profile_defaults_2 = modulemd_ordered_str_keys (v1_self_2->profile_defaults, modulemd_strcmp_sort);
+
+  for (i = 0; i < profile_defaults_1->len; i++)
```

Comment:
Hmm, I just realized that the implementation of `modulemd_hash_table_sets_are_equal()` is identical to this (because we only need to compare the keys for that). Probably best if we just add a `#define modulemd_hash_table_keys_are_equal modulemd_hash_table_sets_are_equal()` in private/modulemd-util.h and use that here instead. No sense duplicating the code.